knowledge-base/grid-search-column-menu.md

```diff
@@ -0,0 +1,168 @@
+    private TelerikGrid<SampleModel>? GridRef { get; set; }
+
+    private List<SampleModel> GridData { get; set; } = new();
+
+    // Non-string columns should not take part in the custom logic
+    private readonly List<string> GridStringFields = new List<string>() { nameof(SampleModel.Name), nameof(SampleModel.Description) };
+
+    private async Task OnGridStateChanged(GridStateEventArgs<SampleModel> args)
+    {
+        // This will be true also for column resizing, reordering and locking.
+        // Some additional checks exist below.
+        if (args.PropertyName == "ColumnStates")
+        {
+            var searchFilterDescriptors = ((CompositeFilterDescriptor)args.GridState.SearchFilter).FilterDescriptors;
+
+            if (searchFilterDescriptors.Any())
+            {
+                var searchValue = ((FilterDescriptor)searchFilterDescriptors.First()).Value;
+                bool shouldRebindGrid = false;
+
+                var visibleStringColumnFields = new List<string>();
+                var filterDescriptorsToRemove = new List<IFilterDescriptor>();
+
+                foreach (GridColumnState colState in args.GridState.ColumnStates)
+                {
+                    if (!string.IsNullOrEmpty(colState.Field) &&
+                        GridStringFields.Contains(colState.Field) &&
+                        (!colState.Visible.HasValue || colState.Visible.Value))
+                    {
+                        visibleStringColumnFields.Add(colState.Field);
+                    }
+                }
+
+                // Find FilterDescriptors for hidden columns
+                foreach (FilterDescriptor fd in searchFilterDescriptors)
+                {
+                    if (!visibleStringColumnFields.Contains(fd.Member))
+                    {
+                        filterDescriptorsToRemove.Add(fd);
+                    }
+                }
+                // Remove FilterDescriptors for hidden columns
+                foreach (FilterDescriptor fd in filterDescriptorsToRemove)
+                {
+                    searchFilterDescriptors.Remove(fd);
+                    shouldRebindGrid = true;
+                }
+
+                // Add FilterDescriptors for newly shown columns
+                foreach (string field in visibleStringColumnFields)
+                {
+                    if (!searchFilterDescriptors.Any(x => ((FilterDescriptor)x).Member == field))
+                    {
+                        searchFilterDescriptors.Add(new FilterDescriptor()
+                        {
+                            Member = field,
+                            MemberType = typeof(string),
+                            Operator = FilterOperator.Contains,
+                            Value = searchValue
+                        });
+                        shouldRebindGrid = true;
+                    }
+                }
+
+                // Apply the changes in args.GridState.SearchFilter and rebind the Grid.
```

Comment:
Note how this code comment has ending punctuation while other code comments don't. Ideally, the use of ending punctuation should be consistent (I don't have strict preferences in either direction).